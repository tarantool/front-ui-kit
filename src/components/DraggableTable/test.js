import React from 'react';
import renderer from 'react-test-renderer';
import { DraggableTable } from './index';

const noop = () => void 0;

it('DraggableTable renders correctly', () => {
  const columns = [
    {
      header: 'Field',
      accessor: 'field',
    },
    {
      header: 'Type',
      accessor: 'type',
    },
  ];

  const data = Array.apply(null, { length: 10 }).map((_, index) => ({
    id: index,
    uuid: index,
    field: `field ${index}`,
    type: `type ${index}`,
    selected: index % 2,
  }));

  const tree = renderer
    .create(
      <DraggableTable
        columns={columns}
        data={data}
        onChange={noop}
        onSelectRow={noop}
        withPositionCol
        draggableOnlySelected
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
