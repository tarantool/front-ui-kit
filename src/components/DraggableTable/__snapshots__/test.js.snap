// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableTable renders correctly 1`] = `
<table
  className="css-1cdxkp1"
>
  <thead>
    <tr
      className="css-lfnnak"
    >
      <th
        className="css-lekys9"
      />
      <th
        className="css-lekys9"
      />
      <th
        className="css-a3r429"
      >
        Field
      </th>
      <th
        className="css-a3r429"
      >
        Type
      </th>
      <th
        className="css-1ljr3r4"
      >
        â„–
      </th>
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className="css-18qxd7n"
    >
      <td
        className="css-lekys9"
      >
        <svg
          className="css-1fn93nz"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
      </td>
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={1}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox-checked"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 1
      </td>
      <td
        className="css-17z4cjy"
      >
        type 1
      </td>
      <td
        className="css-1mppsjo"
      >
        1
      </td>
    </tr>
    <tr
      className="css-18qxd7n"
    >
      <td
        className="css-lekys9"
      >
        <svg
          className="css-1fn93nz"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
      </td>
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={1}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox-checked"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 3
      </td>
      <td
        className="css-17z4cjy"
      >
        type 3
      </td>
      <td
        className="css-1mppsjo"
      >
        2
      </td>
    </tr>
    <tr
      className="css-18qxd7n"
    >
      <td
        className="css-lekys9"
      >
        <svg
          className="css-1fn93nz"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
      </td>
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={1}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox-checked"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 5
      </td>
      <td
        className="css-17z4cjy"
      >
        type 5
      </td>
      <td
        className="css-1mppsjo"
      >
        3
      </td>
    </tr>
    <tr
      className="css-18qxd7n"
    >
      <td
        className="css-lekys9"
      >
        <svg
          className="css-1fn93nz"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
      </td>
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={1}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox-checked"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 7
      </td>
      <td
        className="css-17z4cjy"
      >
        type 7
      </td>
      <td
        className="css-1mppsjo"
      >
        4
      </td>
    </tr>
    <tr
      className="css-18qxd7n"
    >
      <td
        className="css-lekys9"
      >
        <svg
          className="css-1fn93nz"
          viewBox="0 0 16 16"
        >
          <use
            xlinkHref="#burger"
          />
        </svg>
      </td>
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={1}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox-checked"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 9
      </td>
      <td
        className="css-17z4cjy"
      >
        type 9
      </td>
      <td
        className="css-1mppsjo"
      >
        5
      </td>
    </tr>
    <tr
      className="css-lfnnak"
    >
      <td
        className="css-lekys9"
      />
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={0}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 0
      </td>
      <td
        className="css-17z4cjy"
      >
        type 0
      </td>
      <td
        className="css-1mppsjo"
      />
    </tr>
    <tr
      className="css-lfnnak"
    >
      <td
        className="css-lekys9"
      />
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={0}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 2
      </td>
      <td
        className="css-17z4cjy"
      >
        type 2
      </td>
      <td
        className="css-1mppsjo"
      />
    </tr>
    <tr
      className="css-lfnnak"
    >
      <td
        className="css-lekys9"
      />
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={0}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 4
      </td>
      <td
        className="css-17z4cjy"
      >
        type 4
      </td>
      <td
        className="css-1mppsjo"
      />
    </tr>
    <tr
      className="css-lfnnak"
    >
      <td
        className="css-lekys9"
      />
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={0}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 6
      </td>
      <td
        className="css-17z4cjy"
      >
        type 6
      </td>
      <td
        className="css-1mppsjo"
      />
    </tr>
    <tr
      className="css-lfnnak"
    >
      <td
        className="css-lekys9"
      />
      <td
        className="css-lekys9"
      >
        <label
          className="css-759u60 meta__checkbox_class"
        >
          <input
            checked={0}
            className="css-qjlde6"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="css-79elbk"
          >
            <svg
              className="css-1v31moo"
              viewBox="0 0 16 16"
            >
              <use
                xlinkHref="#checkbox"
              />
            </svg>
          </div>
        </label>
      </td>
      <td
        className="css-17z4cjy"
      >
        field 8
      </td>
      <td
        className="css-17z4cjy"
      >
        type 8
      </td>
      <td
        className="css-1mppsjo"
      />
    </tr>
  </tbody>
</table>
`;
